### PR TITLE
Various API fixes

### PR DESCRIPTION
**Pull request by IRacle1**
Exactly what I changed:
- unnecessary `#pragma` removed
- few comments in the transpalers are corrected
- fixed `ExplodingGrenadeEventArgs.TargetsToAffect` modif, when frag grenade explode
- deleted cringe `Map.random`
- better `IEnumerable.ToString` in `StringExtensions`
- Fixed `Respawn.PlayEffects`
- and other little changes

### CODE REVIEW

File: Exiled.API/Features/Attributes/CustomItemAttribute.cs [hunk 1]
```diff
@@ -7,7 +7,6 @@
 
 namespace Exiled.API.Features.Attributes
 {
-#pragma warning disable 1584
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Attributes/CustomItemAttribute.cs:10]: Do not remove them or @joker-119 will cry.

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Attributes/CustomItemAttribute.cs (hunk 1 above)
Comment (IRacle1) [reply Exiled.API/Features/Attributes/CustomItemAttribute.cs:10]: I'm ready to make a sacrifice

Comment (NaoUnderscore) [reply Exiled.API/Features/Attributes/CustomItemAttribute.cs:10]: And I'm ready to not approve nor merge it unless you solve what I just pointed out.

Comment (IRacle1) [reply Exiled.API/Features/Attributes/CustomItemAttribute.cs:10]: And I only need to revert warning 1584?

Comment (IRacle1) [reply Exiled.API/Features/Attributes/CustomItemAttribute.cs:10]: Okay, i'll revert it tomorrow

Comment (NaoUnderscore) [reply Exiled.API/Features/Attributes/CustomItemAttribute.cs:10]: mmk

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -123,7 +123,7 @@ public Player(GameObject gameObject)
         /// <summary>
         /// Gets a <see cref="Dictionary{TKey, TValue}"/> containing all <see cref="Player"/>'s on the server.
         /// </summary>
-        public static Dictionary<GameObject, Player> Dictionary { get; } = new(20);
+        public static Dictionary<GameObject, Player> Dictionary { get; } = new(20, new ReferenceHub.GameObjectComparer());
```
Comment (IRacle1) [inline Exiled.API/Features/Player.cs:126]: As I remember, @Undid-Iridium Told me about storing gameObjects in memory, and he was right, the usual check (at least for null) does not work with gameObjects, and I saw that NWs use this TypeComparer in their dictionary, so I think it's better

**Review by BuildBoy12 — APPROVED** [review]

**Review by NaoUnderscore — APPROVED** [review]

**Pull request merged**